src/test/scala/io/iohk/ethereum/vm/DataWordSpec.scala

```diff
@@ -100,9 +106,57 @@ class DataWordSpec extends FunSuite with PropertyChecks {
         assert(DataWord(n1) / DataWord(n2) == DataWord(n1 / n2))
       }
     }
-    assertThrows[ArithmeticException] {
-      DataWord(1) / DataWord(0)
+    forAll(specialCases) {(n1: BigInt, n2: BigInt) =>
+      whenever(n1 > 0 && n2 > 0) {
+        assert(DataWord(n1) / DataWord(n2) == DataWord(n1 / n2))
+      }
+    }
+    assert(DataWord(1) / Zero == Zero)
+  }
+
+  test("sdiv") {
+    forAll(bigIntGen, bigIntGen) {(n1: BigInt, n2: BigInt) =>
+      whenever(n2 != 0) {
+        val expected: BigInt = toUnsignedBigInt(toSignedBigInt(n1) / toSignedBigInt(n2))
+        assert((DataWord(n1) sdiv DataWord(n2)) == DataWord(expected))
+      }
+    }
+    assert((DataWord(-1) sdiv DataWord(-MaxValue.toBigInt)) == DataWord(-1))
+    assert((DataWord(-1) sdiv Zero) == Zero)
+  }
+
+  test("mod") {
+    forAll(bigIntGen, bigIntGen) {(n1: BigInt, n2: BigInt) =>
+      whenever(n2 != 0) {
+        assert((DataWord(n1) mod DataWord(n2)) == DataWord(n1 mod n2))
+      }
     }
+    assert((DataWord(-1) mod DataWord(MaxValue.toBigInt)) == Zero)
+    assert((DataWord(1) mod Zero) == Zero)
+  }
+
+  test("smod") {
+    assert((DataWord(Modulus - 1) smod DataWord(3)) == DataWord(Modulus - 1))
+    assert((DataWord(-1) smod DataWord(MaxValue.toBigInt)) == Zero)
+    assert((DataWord(1) smod Zero) == Zero)
```

Comment:
`DataWord(Modulus - 1` and `DataWord(MaxValue.toBigInt)` are simply equivalent to `MaxValue`.
Also this could test more cases.